dali/pipeline/operator/arg_helper.h

```diff
@@ -15,96 +15,95 @@
 #ifndef DALI_PIPELINE_OPERATOR_ARG_HELPER_H_
 #define DALI_PIPELINE_OPERATOR_ARG_HELPER_H_
 
-#include <dali/pipeline/operator/argument.h>
-#include <dali/pipeline/operator/op_spec.h>
-#include <dali/pipeline/data/tensor.h>
-#include <dali/pipeline/data/views.h>
 #include <memory>
 #include <string>
+#include <utility>
+#include <vector>
+#include "dali/pipeline/data/tensor.h"
+#include "dali/pipeline/data/views.h"
+#include "dali/pipeline/operator/argument.h"
+#include "dali/pipeline/operator/op_spec.h"
 
 namespace dali {
 
-template <typename T>
+template <typename T, int ndim = 0>
 class ArgValue {
  public:
-  ArgValue() = default;
-  ArgValue(ArgValue &&) = default;
-  inline ArgValue(const ArgValue &other) { *this = other; }
-  explicit inline ArgValue(const std::string &name) : name_(name) {
-  }
+  static_assert(ndim == 0 || ndim == 1,
+    "Only Scalar and 1D arguments are supported at the moment");
 
-  void Update(const OpSpec &spec, ArgumentWorkspace &ws, bool use_default = true) {
-    if (spec.HasTensorArgument(name_)) {
-      arg_input_ = &ws.ArgumentInput(name_);
-      gpu_dirty_ = true;
-      is_set_ = true;
-    } else {
-      is_set_ = spec.HasArgument(name_);
-      arg_input_ = nullptr;
-      if (use_default || is_set_)
-        value_ = spec.GetArgument<T>(name_);
-    }
+  using TLV = TensorListView<StorageCPU, const T, ndim>;
+  using TV = TensorView<StorageCPU, const T, ndim>;
+
+  ArgValue(const std::string &arg_name, const OpSpec &spec)
+      : arg_name_(arg_name),
+        has_arg_const_(spec.HasArgument(arg_name)),
+        has_arg_input_(spec.HasTensorArgument(arg_name)) {
```

Comment:
...if you follow the `move`, then:
```suggestion
        has_arg_const_(spec.HasArgument(arg_name_)),
        has_arg_input_(spec.HasTensorArgument(arg_name_)) {
```